### ISSUE TITLE
Security alerts to messaging (ntfy, telegram...)

### ISSUE THREAD

**Issue by Raito00** [Ylianst/MeshCentral#6495]
**Is your feature request related to a problem? Please describe.**
Meshcentral have great Messaging solution like Ntfy, telegram ... get notifications about devices and login token but dont have security alerts. Like alerts: Invalid password and Invalid 2FA

![image](https://github.com/user-attachments/assets/a17ef201-7357-44b2-aa0b-5f98e8824dba)

**Describe the solution you'd like**
Add [Notification Settings] for Messaging (ntfy, telegram ...) security alerts option - "Invalid password", "Invalid 2FA"

**Additional context**
![image](https://github.com/user-attachments/assets/f94ee085-6037-4fef-9e23-fb73a3761c00)

**Comment by Raito00** [comment]
@si458 Could this additional feature be considered for a future update?

**Comment by Br0kenSilos** [comment]
Agree that this might be a nice feature to add.

**Comment by si458** [comment]
just looking into this as its quite easy
we already have a messaging system in place for the different messaging method a user can use
would you want the notifications by email ALSO or just the messaging method the user has setup?
me for example i use pushover

**Comment by Br0kenSilos** [comment]
Honestly if it were me, I would love to be able to have a choice of either or both. I feel like most people would feel the same way.

Good question about the notification flood... I think your idea is probably better because launching that many emails is a quick way to get the sending email account banned / blacklisted. So I would say doing it the way the device connects work is a better option. I guess, a perfect solution here would be to make it configurable. Maybe a notification rate limiter of some kind?

**Comment by si458** [comment]
@Br0kenSilos my idea was to count the number of failed attempts etc within the 60 seconds, then at every minute like the device notification, it will send u a notification saying `failed attempt X number of times from these ip addresses` and `failed 2fa attempts Y number of times from these ip addresses` and overally if nothing happen then a single notification saying both :)

As for email or messaging, yes I agree make it configurable so the end user can pick between email or messaging or both!

**Comment by Br0kenSilos** [comment]
Agree on both. However on the first thing you state "overall if nothing happen then a single notification saying both". Does that mean sending a notification every minute stating nothing has happened and all is quiet? Just clarifying what you mean. Because if all is quiet, I would assume no notifications at all would be the default.

**Comment by si458** [comment]
@Br0kenSilos yes sorry, if nothing happens, no notifications then if say 3 attempts for password and 4 for 2fa then 1 single notification saying "3 attempts for password and 4 attempts for 2fa" if that makes sense, then at the 60seconds mark the counter gets reset back to 0, until the next time it happens,

U can already do like a cooloff period for failed logins so hopefully u would use that mixed with this notification method to at least be alerted that somebody's trying to get into ur account

**Comment by Br0kenSilos** [comment]
Ah. Yes. I understand. That seems perfect to me.
Yes, I use the cool off period in my config as well.

**Comment by Raito00** [comment]
This feature coming soon?

**Comment by si458** [comment]
@Raito00 it's on the to-do list
